### PR TITLE
Remove redundant type in `@param` occurrences

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

This decision was stated in many places, [here](https://github.com/symfony/symfony/pull/54489#discussion_r1553560582) is the most recent (I guess).

A Rector rule did the job, however it was not accepted in the upstream:

- https://github.com/rectorphp/rector-src/pull/5811

Still I applied it + PHP CS Fixer `phpdoc_align` with `vertical` to get this result. Finally, an issue has been reported to Jetbrains so PHPStorm may supports this syntax: https://youtrack.jetbrains.com/issue/WI-76947/Allow-param-to-omit-the-type. 🤞

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php [hunk 1]
```diff
@@ -22,16 +22,16 @@ class MapEntity extends ValueResolver
 {
     /**
      * @param class-string|null          $class         The entity class
-     * @param string|null                $objectManager Specify the object manager used to retrieve the entity
-     * @param string|null                $expr          An expression to fetch the entity using the {@see https://symfony.com/doc/current/components/expression_language.html ExpressionLanguage} syntax.
-     *                                                  Any request attribute are available as a variable, and your entity repository in the 'repository' variable.
+     * @param                            $objectManager Specify the object manager used to retrieve the entity
+     * @param                            $expr          An expression to fetch the entity using the {@see https://symfony.com/doc/current/components/expression_language.html ExpressionLanguage} syntax.
+     *                                                 Any request attribute are available as a variable, and your entity repository in the 'repository' variable.
```
Comment (GromNaN) [inline src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php]:
```suggestion
     *                                                  Any request attribute are available as a variable, and your entity repository in the 'repository' variable.
```

Comment (alexandre-daubois) [reply src/Symfony/Bridge/Doctrine/Attribute/MapEntity.php]: Good catch, thanks!

**Comment by derrabus** [comment]
How's tool support for this syntax these days?

**Comment by alexandre-daubois** [comment]
- PHPStorm struggles with this, hope my issue in Jetbrains' Youtrack gets some attention!
- Psalm seems ok, as it didn't complain in this PR
- PHPStan handles it well: https://phpstan.org/r/e471345c-3f0b-4a71-a593-c72639017700
- phpDocumentor defines the type as optional so it's also fine
- VSCode seems also fine with it

Am I missing something? I'd be happy to test more!

**Comment by nicolas-grekas** [comment]
Please allow me to close here as this would need some automation to enforce in the long run. At the moment, this isn't worth the review+conflict-resolution time this will create IMHO.

**Comment by alexandre-daubois** [comment]
Fair enough 👍

**Pull request closed without merging**